pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -640,8 +640,60 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	if cd.Status.ProvisionRef == nil {
 		if cd.Status.InstallRestarts > 0 && cd.Annotations[tryInstallOnceAnnotation] == "true" {
 			cdLog.Debug("not creating new provision since the deployment is set to try install only once")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.ProvisionStoppedCondition,
+				corev1.ConditionTrue,
+				"InstallOnlyOnceSet",
+				"Deployment is set to try install only once",
+				controllerutils.UpdateConditionIfReasonOrMessageChange)
+			if changed {
+				cd.Status.Conditions = conditions
+				cdLog.Debugf("setting ProvisionStoppedCondition to %v", corev1.ConditionTrue)
+				if err := r.Status().Update(context.TODO(), cd); err != nil {
+					cdLog.WithError(err).Log(controllerutils.LogLevel(err), "failed to update cluster deployment status")
+					return reconcile.Result{}, err
+				}
+			}
+			return reconcile.Result{}, nil
+		}
+		if cd.Spec.InstallAttemptsLimit != nil && cd.Status.InstallRestarts >= int(*cd.Spec.InstallAttemptsLimit) {
+			cdLog.Debug("not creating new provision since the install attempts limit has been reached")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.ProvisionStoppedCondition,
+				corev1.ConditionTrue,
+				"InstallAttemptsLimitReached",
+				"Install attempts limit reached",
```

Comment:
not sure what this value represents, but if this is the message of the condition, i think it makes sense to include for info for the user like `"Install attempts limit of %d reached", cd.Spec.InstallAttemptsLimit` which helps them figure out why the limit was reached.

Reply:
This was removed after previous review feedback as we thought exact limit is already available in `cd.Spec`.

Reply:
So conditions can and are used by clients outside the context of the complete definition of the cluster deployment esp error conditions,

for example, when the cloud providers return errors for quota exceeded, even though the quota is available to be looked at in the console, the error message usually goes something like `requested 8 vCPUs but only 4 available in the region` . So personally I think the messages for these conditions should be useful/actionable for the users.

but i'll defer to previous review.